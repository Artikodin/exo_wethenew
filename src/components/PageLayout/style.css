.header {
  height: 60px;
  background-color: #ffe299;
  box-shadow: 0px 6px 25px -19px rgba(150,150,150,1);
  display: flex;
}

.header > div {
  margin: auto 0 auto 10px;
  font-weight: 700;
  font-size: 22px;
}

.content {
  max-width: 1100px;
  width: 100%;
  margin: auto;
}

.mainContainer {
  padding: 50px 0 0 0;
  height: calc(100vh - 110px);
  overflow: auto;
  background-color: #ffecc1;
}
