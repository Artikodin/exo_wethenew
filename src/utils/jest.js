import React from "react";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router, Route } from "react-router-dom";

/**
 * renderWithRouter
 *
 * Render a component with a mocked router. 
 * Usefull when you want to test a component containing react router component.
 * 
 * @param {node} children - react component you want to add routing logic
 * @param {object} routeParams - different parameter you use to simulate a routing behaviour
 *
 * @return {node} A react router mock
 */

export const renderWithRouter = (
  children,
  {
    url = "/",
    route: { path = "/", isExact = false } = {},
    history = createMemoryHistory({ initialEntries: [url] }),
  } = {}
) => {
  return {
    ...render(
      <Router history={history}>
        <Route path={path} exact={isExact}>
          {children}
        </Route>
      </Router>
    ),
    history,
  };
};
