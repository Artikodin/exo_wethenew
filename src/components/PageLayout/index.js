import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

import "./style.css";

/**
 * PageLayout
 *
 * This components is in charge to display the page layout
 * to keep same design pattern through the project
 *
 * @return {jsx}
 */

const PageLayout = ({ children }) => {
  return (
    <div>
      <header className="header">
        <div>
          <Link to="/" className="custom-link">
            WETHENEW
          </Link>
        </div>
      </header>
      <div className="mainContainer">
        <div className="content">{children}</div>
      </div>
    </div>
  );
};

const { node } = PropTypes;
PageLayout.propTypes = { children: node };
PageLayout.defaultProps = { children: null };

export default PageLayout;
