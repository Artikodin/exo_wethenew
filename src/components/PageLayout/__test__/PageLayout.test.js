import React from "react";
import { cleanup } from "@testing-library/react";
import { renderWithRouter } from "../../../utils/jest";

import PageLayout from "..";

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

describe("<PageLayout />", () => {
  test("should render withouth a problem", () => {
    const { container } = renderWithRouter(<PageLayout></PageLayout>);

    expect(container.firstChild).toMatchSnapshot();
  });

  test("should contain correct children", () => {
    const children = "test";
    const { getByText } = renderWithRouter(<PageLayout>{children}</PageLayout>);

    const dropdownNode = getByText(children);
    expect(dropdownNode).toBeInTheDocument();
  });
});
