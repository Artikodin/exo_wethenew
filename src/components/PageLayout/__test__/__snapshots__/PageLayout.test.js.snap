// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLayout /> should contain correct children 1`] = `
<div>
  <header
    class="header"
  >
    <div>
      <a
        class="custom-link"
        href="/"
      >
        WETHENEW
      </a>
    </div>
  </header>
  <div
    class="mainContainer"
  >
    <div
      class="content"
    />
  </div>
</div>
`;

exports[`<PageLayout /> should render withouth a problem 1`] = `
<div>
  <header
    class="header"
  >
    <div>
      <a
        class="custom-link"
        href="/"
      >
        WETHENEW
      </a>
    </div>
  </header>
  <div
    class="mainContainer"
  >
    <div
      class="content"
    />
  </div>
</div>
`;

exports[`<PageLayout /> should render withouth a problem 2`] = `
<div>
  <header
    class="header"
  >
    <div>
      <a
        class="custom-link"
        href="/"
      >
        WETHENEW
      </a>
    </div>
  </header>
  <div
    class="mainContainer"
  >
    <div
      class="content"
    />
  </div>
</div>
`;
